sycl/doc/dev/RuntimeTestingGuidelines.md

```diff
@@ -0,0 +1,123 @@
+# SYCL Runtime and Device Headers Testing Guidelines
+
+## SYCL Host Runtime Testing
+
+Most changes to the SYCL Host Runtime are device-agnostic, thus can and must be
+tested in isolation from the rest of the SYCL toolchain. Below is the
+information on some techniques, that allow one to isolate these parts of the
+runtime and simulate different behavior of low-level runtimes.
+
+### Plugin Interface Mock
+
+PI Mock layer provides capabilities to override plugins' behavior. There are two
```

Comment:
There are 2 options here. Could you add a recommendation on what path to go in different cases (if there is any difference)?